qt_dotgraph/src/qt_dotgraph/dot_to_qt.py

```diff
@@ -256,15 +260,28 @@ def dotcode_to_qt_items(self, dotcode, highlight_level, same_label_siblings=Fals
         #graph = pygraphviz.AGraph(string=self._current_dotcode, strict=False, directed=True)
         #graph.layout(prog='dot')
 
+        nodes = self.parse_nodes(graph, highlight_level)
+        edges = self.parse_edges(graph, nodes, highlight_level, same_label_siblings)
+        return nodes, edges
+
+    def parse_nodes(self, graph, highlight_level):
+        '''
+        Recursively searches all nodes inside the graph and all subgraphs
+        '''
+
         # let pydot imitate pygraphviz api
         graph.nodes_iter = graph.get_node_list
-        graph.edges_iter = graph.get_edge_list
-
         graph.subgraphs_iter = graph.get_subgraph_list
 
+        num_subgraphs = len(graph.get_subgraph_list())
+
         nodes = {}
         for subgraph in graph.subgraphs_iter():
             subgraph_nodeitem = self.getNodeItemForSubgraph(subgraph, highlight_level)
+            num_subgraphs = len(subgraph.get_subgraph_list())
+
+            nodes = dict(nodes, **self.parse_nodes(subgraph, highlight_level))
```

Comment:
I think the following would be more readable and make the intention of the line clearer:

```
nodes.update(...)
```

Same below for `edges`.